### ISSUE TITLE
IronPython reading C# documentation xml problem

### ISSUE THREAD

**Issue by refaelsh** [IronLanguages/main#1435]
IronPython cant read C#'s assembly xml document file correctly. If it contains methods with "out/ref" parameters it confuses IronPython. Steps to reproduce:
Lets say You have the following C# assembly:

```
    /// <summary>
    /// Some description1.
    /// </summary>
    public class clsBar
    {
        /// <summary>
        /// Some description2.
        /// </summary>
        public static String strFoo= "foo";

        /// <summary>
        /// Some description3.
        /// </summary>
        public clsBar()
        {

        }

        /// <summary>
        /// Some description4.
        /// </summary>
        public Int32 someMethod1()
        {
            return 8;
        }

        /// <summary>
        /// Some description5.
        /// </summary>
        public Int32 someMethod2(String strSome)
        {
            return 8;
        }

        /// <summary>
        /// Some description6.
        /// </summary>
        public Int32 someMethod3(out String strSome)
        {
            strSome = "Some string.";
            return 8;
        }
    }
```

I've loaded this assembly in IronPython and created an instance like this:
`dcl = clsBar()`
Running help(dcl.someMethod1) and help(dcl.someMethod2) produced the correct results:

```
help(dcl.someMethod1)
Help on built-in function someMethod1

 |  someMethod1(...)
 |          someMethod1(self: DCLoadWrapperPython) -> int  |            |              Some description4.  |
```

```
help(dcl.someMethod2)
Help on built-in function someMethod2

 |  someMethod2(...)
 |          someMethod2(self: DCLoadWrapperPython, strSome: str) -> int  |            |              Some description5.  |          

```

And now here is the problem, running help(dcl.someMethod3) produces the wrong result:

```
help(dcl.someMethod3)
Help on built-in function someMethod3

 |  someMethod3(...)
 |          someMethod3(self: DCLoadWrapperPython) -> (int, str)  |   
```

It no longer shows the summary tag from C#'s code (or more exactly from the xml file documentation that was generated by Visual Studio alongside the assembly).
The only difference between someMethod2 and someMethod3 is that strSome parameter is an "out" parameter in someMethod3 --> it confuses IronPython.

P.S.
Also, help(dcl.strFoo) does not works.

**Comment by slide** [comment]
How did you generate the xml documentation? I just want to make sure I have the same steps as you before trying to debug this.

**Comment by refaelsh** [comment]
I used the default way that Visual Studio provides:
1. I add the "///" to every method I want to document.
2. In Project Properties -> Build Tab I checked the "XML documentation file" checkbox.

From here, Visual Studio automatically generates the xml file.

**Comment by slide** [comment]
I tried this same thing, but none of the docs show up.

**Comment by refaelsh** [comment]
To @slide: This is strange. I dont know what to tell You.

**Comment by slide** [comment]
I'll continue looking at it.

**Comment by slide** [comment]
Ok, I figured out what my issue was. I can now get the docs like you were, and I even fixed a bug (if there is no namespace on the class, then you wouldn't get the docs). I'll debug why the out/ref stuff causes issues now.

**Comment by refaelsh** [comment]
Thank You very much @slide :-)

**Comment by refaelsh** [comment]
Thank You @slide very much for fixing this issue :-)
It helps Me a lot!

One more question please: I understand that the fix is labeled to be in version 2.7.6.4, when it will be available as an update to Visual Studio NuGet package (current version in NuGet is 2.7.5)?

**Comment by slide** [comment]
I labelled it that way because that will possibly be the next released
version.
